File: benchmark/lua/factorial_recursive_approximation.lua
-- Note: Lua uses float for large numbers
local start = os.clock()
local function fact(n)
    if n <= 1 then return 1.0 end
    return n * fact(n - 1)
end
local result = fact(150)
local finish = os.clock()
print("fact(150) = " .. result)
print(string.format("Time: %.0fms", (finish - start) * 1000))

File: benchmarks/src/perl/factorial_recursive_approximation.pl
use Time::HiRes qw(time);
sub fact {
    my ($n) = @_;
    return 1.0 if $n <= 1;
    return $n * fact($n - 1);
}
my $start = time * 1000;
my $res = fact(150);
my $end = time * 1000;
print "fact(150) = $res
";
printf "Time: %.0fms
", $end - $start;